src/input_context.cpp

```diff
@@ -606,244 +739,95 @@ action_id input_context::display_menu( const bool permit_execute_action )
                    && permit_execute_action ) {
-            if( !filtered_registered_actions.empty() ) {
-                status = s_execute;
-            }
-        } else if( action == "DOWN" ) {
-            if( !filtered_registered_actions.empty()
-                && filtered_registered_actions.size() > display_height
-                && scroll_offset < filtered_registered_actions.size() - display_height ) {
-                scroll_offset++;
-            }
-        } else if( action == "UP" ) {
-            if( !filtered_registered_actions.empty()
-                && scroll_offset > 0 ) {
-                scroll_offset--;
-            }
-        } else if( action == "PAGE_DOWN" || action == "SCROLL_DOWN" ) {
-            if( filtered_registered_actions.empty() ) {
-                // do nothing
-            } else if( scroll_offset + display_height < filtered_registered_actions.size() ) {
-                scroll_offset += std::min( display_height, filtered_registered_actions.size() -
-                                           display_height - scroll_offset );
-            } else if( filtered_registered_actions.size() > display_height ) {
-                scroll_offset = 0;
-            }
-        } else if( action == "PAGE_UP" || action == "SCROLL_UP" ) {
-            if( filtered_registered_actions.empty() ) {
-                // do nothing
-            } else if( scroll_offset >= display_height ) {
-                scroll_offset -= display_height;
-            } else if( scroll_offset > 0 ) {
-                scroll_offset = 0;
-            } else if( filtered_registered_actions.size() > display_height ) {
-                scroll_offset = filtered_registered_actions.size() - display_height;
+            if( !kb_menu.filtered_registered_actions.empty() ) {
+                kb_menu.status = kb_menu.s_execute;
             }
+        } else if( action == "PAGE_UP" || action == "PAGE_DOWN" || action == "HOME" || action == "END" ) {
+            continue; // do nothing - on tiles version for some reason this counts as pressing various alphabet keys
+        } else if( action == "TEXT.CLEAR" ) {
+            kb_menu.filter_text.assign( "" );
+            kb_menu.filtered_registered_actions = filter_strings_by_phrase( org_registered_actions,
+                                                  "" );
+        } else if( !kb_menu.get_is_open() ) {
+            break;
         } else if( action == "QUIT" ) {
-            if( status != s_show ) {
-                status = s_show;
+            if( kb_menu.status != kb_menu.s_show ) {
+                kb_menu.status = kb_menu.s_show;
             } else {
                 break;
             }
+        } else if( action == "TEXT.INPUT_FROM_FILE" ) {
+            std::string entered = get_input_string_from_file();
+            utf8_wrapper wrap( kb_menu.filter_text );
+            wrap.append( utf8_wrapper( entered.c_str() ) );
+            kb_menu.filter_text += wrap.str();
```

Comment:
This all boils down to just `kb_menu.filter_text += entered;`. Nothing about utf8_wrapper is relevant in this snippet.
```suggestion
            kb_menu.filter_text += get_input_string_from_file();
```

Reply:
got it. still learning how utf8 works with string arrays.